modelarmor/modelarmor_test.go

```diff
@@ -162,6 +275,119 @@ func testSDPTemplate(t *testing.T, projectID string, locationID string) (string,
+				SdpConfiguration: &modelarmorpb.SdpFilterSettings_BasicConfig{
+					BasicConfig: &modelarmorpb.SdpBasicConfig{
+						FilterEnforcement: modelarmorpb.SdpBasicConfig_ENABLED,
+					},
+				},
+			},
+		},
+	}
+
+	req := &modelarmorpb.CreateTemplateRequest{
+		Parent:     fmt.Sprintf("projects/%s/locations/%s", tc.ProjectID, locationID),
+		TemplateId: templateID,
+		Template:   template,
+	}
+
+	response, err := client.CreateTemplate(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create template: %w", err)
+	}
+
+	return response, nil
+}
+
+// testModelArmorAdvancedSDPTemplate creates Model Armor template with Advanced SDP configuration.
+func testModelArmorAdvancedSDPTemplate(t *testing.T, templateID string) (*modelarmorpb.Template, error) {
+	tc := testutil.SystemTest(t)
+
+	projectID := tc.ProjectID
+	locationID := testLocation(t)
+	inspectResponseName, deidentifyResponseName := testSDPTemplate(t, projectID, locationID)
+	client, ctx := testClient(t)
+
+	// Create template with advanced SDP configuration
+	template := &modelarmorpb.Template{
+		FilterConfig: &modelarmorpb.FilterConfig{
+			RaiSettings: &modelarmorpb.RaiFilterSettings{
+				RaiFilters: []*modelarmorpb.RaiFilterSettings_RaiFilter{
+					{
+						FilterType:      modelarmorpb.RaiFilterType_DANGEROUS,
+						ConfidenceLevel: modelarmorpb.DetectionConfidenceLevel_HIGH,
+					},
+					{
+						FilterType:      modelarmorpb.RaiFilterType_HARASSMENT,
+						ConfidenceLevel: modelarmorpb.DetectionConfidenceLevel_MEDIUM_AND_ABOVE,
+					},
+					{
+						FilterType:      modelarmorpb.RaiFilterType_HATE_SPEECH,
+						ConfidenceLevel: modelarmorpb.DetectionConfidenceLevel_HIGH,
+					},
+					{
+						FilterType:      modelarmorpb.RaiFilterType_SEXUALLY_EXPLICIT,
+						ConfidenceLevel: modelarmorpb.DetectionConfidenceLevel_HIGH,
+					},
+				},
+			},
+			SdpSettings: &modelarmorpb.SdpFilterSettings{
+				SdpConfiguration: &modelarmorpb.SdpFilterSettings_AdvancedConfig{
+					AdvancedConfig: &modelarmorpb.SdpAdvancedConfig{
+						InspectTemplate:    inspectResponseName,
+						DeidentifyTemplate: deidentifyResponseName,
+					},
+				},
+			},
+		},
+	}
+	// Prepare the request for creating the template.
+	req := &modelarmorpb.CreateTemplateRequest{
+		Parent:     fmt.Sprintf("projects/%s/locations/%s", tc.ProjectID, locationID),
+		TemplateId: templateID,
+		Template:   template,
+	}
+
+	// Create the template.
+	response, err := client.CreateTemplate(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create template: %w", err)
+	}
+
+	return response, nil
+
+}
+
+// testCleanupTemplate deletes the specified Model Armor template if it exists,
+// ignoring the error if the template is already deleted.
+func testCleanupTemplate(t *testing.T, templateName string) {
+	t.Helper()
+
+	client, ctx := testClient(t)
+	if err := client.DeleteTemplate(ctx, &modelarmorpb.DeleteTemplateRequest{Name: templateName}); err != nil {
```

Comment:
nit: move the call to `client.DeleteTemplate()` outside of the `if` statement. Doing so makes the code a bit easier to read.

suggestion:

```go
err := client.DeleteTemplate(ctx, &modelarmorpb.DeleteTemplateRequest{Name: templateName})
if err == nil {
    return
}
if terr, ok := grpcstatus.FromError(err), !ok || terr.Code() != grpccodes.NotFound {
    t.Fatalf("testCleanupTemplate: failed to delete template %v", err)
}
```